src/main/webapp/app/course/learning-paths/services/base-api-http.service.ts

```diff
@@ -66,7 +66,22 @@ export abstract class BaseApiHttpService {
         return await this.request<T>(HttpMethod.Get, url, options);
     }
 
-    protected async post<T>(url: string, body?: any): Promise<T> {
-        return await this.request<T>(HttpMethod.Post, url, { body: body });
+    protected async post<T>(
```

Comment:
I just noticed now that you use `private readonly httpClient = inject(HttpClient);` in this file. I think we use constructor injection in every other service. Can we do this here to?

Reply:
inject is a relatively new Angular feature (came with signals) which should replace constructor injection.